core/routes/collection.ts

```diff
@@ -154,18 +154,18 @@ export default function (app: Hono) {
       const body = await c.req.parseBody();
       let name = body._id as string;
 
-      const namePrefix = "changes."
+      const changes = changesToData(body);
       if (!name && collection.nameField) {
         switch (typeof collection.nameField) {
           case "string": {
-            const autoname = body[`${namePrefix}${collection.nameField}`];
+            const autoname = changes[collection.nameField];
```

Comment:
Here it's better to keep the previous code, because it can access to inner properties. For example `body["changes.metas.title"]`.